src/Servers/Kestrel/Transport.MsQuic/src/MsQuicTransportOptions.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic
+{
+    public class MsQuicTransportOptions
+    {
+        /// <summary>
+        /// The maximum number of concurrent bi-directional streams per connection.
+        /// </summary>
+        public ushort MaxBidirectionalStreamCount { get; set; } = 100;
+
+        /// <summary>
+        /// The maximum number of concurrent inbound uni-directional streams per connection.
+        /// </summary>
+        public ushort MaxUnidirectionalStreamCount { get; set; } = 10;
+
+        /// <summary>
+        /// The Application Layer Protocol Negotiation string.
+        /// </summary>
+        public string Alpn { get; set; }
```

Comment:
Note that QUIC requires ALPN.

Reply:
Shouldn't ALPN be a list?

Reply:
Follow up PR. @nibanks do we need to create a session per ALPN, or just a ';' separated list.

Reply:
Currently, MsQuic doesn't support multiple ALPNs for the same session, but it's been on my wish list for some time. It would probably be best to take a `;` separated list (string) to be passed into the Session. It's not high on my priority list right now, but feel free to make a PR to MsQuic if you feel like it.